eth_utils/abi.py

```diff
@@ -42,27 +83,682 @@ def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+
+
+def _get_tuple_type_str_and_dims(s: str) -> Optional[Tuple[str, Optional[str]]]:
+    """
+    Takes a JSON ABI type string.  For tuple type strings, returns the separated
+    prefix and array dimension parts.  For all other strings, returns ``None``.
+    """
+    tuple_type_str_re = "^(tuple)((\\[([1-9]\\d*\b)?])*)??$"
+    match = re.compile(tuple_type_str_re).match(s)
+
+    if match is not None:
+        tuple_prefix = match.group(1)
+        tuple_dims = match.group(2)
+
+        return tuple_prefix, tuple_dims
+
+    return None
+
+
+def abi_to_signature(abi: ABIElement) -> str:
+    """
+    Returns a string signature representation of the function or event ABI
+    and arguments.
+
+    Signatures consist of the name followed by a list of arguments.
+
+    :param abi: Contract ABI.
+    :type abi: `ABI`
+    :return: Stringified ABI signature
+    :rtype: `str`
+
+    .. doctest::
+
+        >>> from eth_utils import abi_to_signature
+        >>> abi = {
+        ...   'constant': False,
+        ...   'inputs': [
+        ...     {
+        ...       'name': 's',
+        ...       'type': 'uint256'
+        ...     }
+        ...   ],
+        ...   'name': 'f',
+        ...   'outputs': [],
+        ...   'payable': False,
+        ...   'stateMutability': 'nonpayable',
+        ...   'type': 'function'
+        ... }
+        >>> abi_to_signature(abi)
+        'f(uint256)'
+    """
+    if abi["type"] == "fallback" or abi["type"] == "receive":
+        name = str(abi["type"])
+        abi_inputs = None
+    elif abi["type"] == "constructor":
+        name = abi["type"]
+        abi_inputs = abi["inputs"]
+    elif abi["type"] == "event":
+        name = abi["name"]
+        abi_inputs = abi["inputs"]
+    else:
+        name = str(abi.get("name", abi["type"]))
+        abi_inputs = abi.get("inputs")
+
+    return f"{name}({_abi_inputs_types(abi_inputs)})"
+
+
+def filter_by_type(type: str, contract_abi: ABI) -> List[ABIElement]:
+    """
+    Return a list of each ``ABIElement`` that is of type ``_type``.
+
+    :param type: Type of ABI element to filter by.
+    :type type: `str`
+    :param contract_abi: Contract ABI.
+    :type contract_abi: `ABI`
+    :return: List of ABI elements of the specified type.
+    :rtype: `List[ABIElement]`
+
+    .. doctest::
+
+        >>> from eth_utils import filter_by_type
+        >>> abi = [
+        ...   {"type": "function", "name": "myFunction", "inputs": [], "outputs": []},
+        ...   {"type": "function", "name": "myFunction2", "inputs": [], "outputs": []},
+        ...   {"type": "event", "name": "MyEvent", "inputs": []}
+        ... ]
+        >>> filter_by_type("function", abi)
+        [{'type': 'function', 'name': 'myFunction', 'inputs': [], 'outputs': []}, \
+{'type': 'function', 'name': 'myFunction2', 'inputs': [], 'outputs': []}]
+    """
+    return [abi for abi in contract_abi if abi["type"] == type]
+
+
+def get_all_function_abis(abi: ABI) -> Sequence[ABIFunction]:
```

Comment:
Same nit here for clarity

```suggestion
def get_all_function_abis(contract_abi: ABI) -> Sequence[ABIFunction]:
```